### PR TITLE
Use IEC standard abbreviations (GiB, TiB, etc)

### LINKED ISSUE #6741: [CT-1911] [Bug] Non-standard abbreviations for kibibyte, etc
### Is this a new bug in dbt-core?

- [X] I believe this is a new bug in dbt-core
- [X] I have searched the existing issues, and I could not find an existing issue for this bug

### Context

This is a copy of https://github.com/dbt-labs/dbt-bigquery/issues/477 since there is an (unused!) copy of the relevant code in dbt-core as well.

### Current Behavior

Described in https://github.com/dbt-labs/dbt-bigquery/issues/477

### Expected Behavior

Described in https://github.com/dbt-labs/dbt-bigquery/issues/477

### Steps To Reproduce

Described in https://github.com/dbt-labs/dbt-bigquery/issues/477

### Relevant log output

_No response_

### Environment

```markdown
- OS:
- Python:
- dbt:
```

### Which database adapter are you using with dbt?

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by dbeatty10**
resolves #6741
equivalent to https://github.com/dbt-labs/dbt-bigquery/pull/482

### Description

When using multiples of 1024, the IEC standard abbreviations are "GiB", "TiB", "PiB" instead of "GB", "TB", "PB" (which are for multiples of 1000).

### More info
Today I Learned (TIL), `format_bytes` (and `format_rows_number`) are defined in dbt-core but redefined in dbt-bigquery and only used in that adapter. So the dbt-core version is unused to best I can tell.

#### Context for the duplication
Removing them from dbt-core was mentioned [here](https://github.com/dbt-labs/dbt-bigquery/pull/48#discussion_r742152754) and [here](https://github.com/dbt-labs/dbt-bigquery/pull/48#discussion_r743485981) as part of the discussion for #48.

It feels to me like one of the following should have happened along with https://github.com/dbt-labs/dbt-bigquery/pull/48:
1. a simultaneous PR should have removed `format_bytes` and `format_rows_number` from dbt-core; or
2. [#48](https://github.com/dbt-labs/dbt-bigquery/pull/48) should have been added to dbt-core instead of dbt-bigquery.

As it stands, the implementations in dbt-core are unused (to the best that we can tell), but would probably need to be maintained to preserve backwards compatibility in case some unknown adapter is using them.

#### Proposal

Going forward, I'd propose that we choose one (and only one) of the following:
1. Keep separate implementations in dbt-core and dbt-bigquery (and fix the KB vs. KiB issue in both)
1. Remove the implementations in dbt-core (so they exist only in dbt-bigquery)
1. Completely move the implementation in dbt-bigquery back into dbt-core

### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] I have [opened an issue to add/update docs](https://github.com/dbt-labs/docs.getdbt.com/issues/new/choose), or docs changes are not required/relevant for this PR
- [x] I have run `changie new` to [create a changelog entry](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

### CODE REVIEW

**Review by ChenyuLInx — APPROVED** [review]
The updated change looks good to me! Learned a bit new things about IEC standard also

File: core/dbt/utils.py [hunk 1]
```diff
@@ -454,7 +454,7 @@ def __get__(self, obj, objtype):
 
 
 def format_bytes(num_bytes):
-    for unit in ["Bytes", "KB", "MB", "GB", "TB", "PB"]:
+    for unit in ["Bytes", "KiB", "MiB", "GiB", "TiB", "PiB"]:
```
Comment (jtcohen6) [inline core/dbt/utils.py:457]: I don't think we actually use this method anywhere in `dbt-core`. It's a holdover from when it was added, for the sole benefit of `dbt-bigquery`, and we forgot to remove it during the Great Adapter Split-Apart of 2021.

What do you think about just removing it?

Comment (dbeatty10) [reply core/dbt/utils.py]: > What do you think about just removing it?

Removed!

I did a quick cursory check for both `format_bytes` and `format_rows_number` in the dbt Labs adapters using a GitHub search like [this](https://github.com/dbt-labs/dbt-redshift/search?q=format_rows_number&type=code). DIdn't see any references other than in dbt-bigquery, which we know about.

I didn't check any non-dbt Labs adapters, although our expectation is that it is not being used. If it _did_ happen to be used by some adapter out there, they'd be on their own to resolve it.

**Review by jtcohen6 — APPROVED** [review]
lgtm aussi

**Comment by dbeatty10** [comment]
Les remerciements @ChenyuLInx and @jtcohen6

Merging now.

**Pull request merged**